src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -127,6 +159,34 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
     this
   }
 
+  /*
+   * Implemented by means of a buffer to keep the last n elements from being returned during iteration.
+   */
+  def dropRight(n: Int): Iterator[A] = {
```

Comment:
You cannot in general re-traverse an iterator, so you can't assume you can "perform two traversals".  Furthermore, even if you can, depending on what it is, it can be very expensive to traverse a second time (e.g. if it's doing some sort of database lookup).  So in the absence of other type information, it has to be buffered.
The problem with making it not on `Iterator` at all, @szeiger, is that you want to make `takeRight` and `dropRight` both be present or not together, and `takeRight` can't be done efficiently on an iterator with anything but a bare array.  But you can't convert to an array first because that isn't lazy.  So you're really stuck: the algorithm needs to be tuned to `Iterator` to be efficient, and there's no other place to put it unless we add a subclass like `BufferedIterator` to isolate the methods that use non-lazy buffering.